mobile/src/viewModels/QoLViewModel.ts

```diff
@@ -0,0 +1,154 @@
+import { createLogger } from 'common/logger';
+import AppController from 'src/controllers';
+import { ILocalSettingsController } from 'src/controllers/LocalSettings';
+import { PartialQol, QolSurveyResults, QolSurveyType } from 'src/constants/QoL';
+import { PersonaArmState } from 'dependencies/persona/lib';
+
+export const logger = createLogger('[QOLModel]');
+
+export default class QOLSurveyViewModel {
+
+    @observable
+    private _questionNum: number;
+    @observable
+    private _domainNum: number;
+    private _surveyResponses: QolSurveyResults;
+    private _armMags: PersonaArmState;
+    public isUnfinished: boolean;
+    public initModel: Promise<void>;
+    public origMags: PersonaArmState;
+    public showInterlude: boolean = false;
+    public QolSurveyType: QolSurveyType;
+    public startDate: number;
+    public questionCompletionDates: number[];
+
+    public readonly numQuestions: number = QUESTIONS_COUNT;
+    public readonly domainQuestionCount: number = DOMAIN_QUESTION_COUNT;
+    private readonly _settings: ILocalSettingsController = AppController.Instance.User.localSettings;
+
+    constructor() {
+        this.initModel = AppController.Instance.User.backend.getPartialQol().then((partialQolState: PartialQol) => {
+            if (partialQolState !== null) {
+                this._questionNum = partialQolState.questionNum;
+                this._domainNum = partialQolState.domainNum;
+                this._surveyResponses = partialQolState.scores;
+                this.startDate = partialQolState.startDate;
+                this.questionCompletionDates = partialQolState.questionCompletionDates;
+                this._armMags = this.getMags(partialQolState.scores);
+                this.isUnfinished = true;
+                this.showInterlude = partialQolState.isFirstTimeQol;
+                return;
+            } else {
+                this.startDate = new Date().getTime();
+                this._questionNum = 0;
+                this._domainNum = 0;
+                const surveyResponses = {};
+                for (let domain of PersonaDomains) {
+                    surveyResponses[domain] = 0;
+                }
+                this._surveyResponses = surveyResponses;
+                this.questionCompletionDates = [];
+                this._armMags = PersonaArmState.createEmptyArmState();
+                this.isUnfinished = false;
+                return;
+            }
+        });
+    }
+
+    async init() {
+        return await this.initModel;
+    }
+
+    @computed
+    get questionNum(): number { return this._questionNum; }
+
+    @computed
+    get domainNum(): number { return this._domainNum; }
+
+    @computed
+    get question(): string { return SurveyQuestions[this._questionNum]; }
+
+    @computed
+    get domain(): string { return PersonaDomains[this._domainNum]; }
+
+    get surveyResponses(): any { return this._surveyResponses; }
+
+    get qolMags(): any { return this._armMags; }
```

Comment:
Can we rename `mag` to `magnitude` or `score` for clarity?

Reply:
done